src/event-cache/EventCache.ts

```diff
@@ -220,12 +225,20 @@ export class EventCache {
             'aws:clientVersion': webClientVersion
         };
 
-        this.events.push({
-            details: JSON.stringify(eventData),
+        const partialEvent = {
             id: v4(),
-            metadata: JSON.stringify(metaData),
             timestamp: new Date(),
             type
+        };
+        this.bus.notify(type, {
```

Comment:
Note: The "Raw" RumEvent is published so that plugins do not have to de-stringify the information they want to look at.

Reply:
question: We are potentially publishing more than needed. Consider the following options.
1. The EventCache publishes whenever any event is recorded.
2. The Resource plugin publishes whenever it records a PerformanceResource event.

Reply:
I can refactor so that record() returns a raw rum event, which includes fields like `id` and `metadata`. That way, ResourcePlugin can publish all the information we need, for instance. 

Do you have any issue with the publishing parsed rum event?

Reply:
Does the web vitals plugin need a reference to the PerformanceResource object? If so, I'm not sure (1) will work.

Reply:
(1) is a simpler implementation

> Does the web vitals plugin need a reference to the PerformanceResource object?

It does not need PerformanceResourceTiming object as a key because the URL is also a good key. But the performance object is the best key

Reply:
+1. I think (1) makes more sense for the points made by @williazz

Reply:
> +1. I think (1) makes more sense for the points made by @williazz

I agree.

> If we go back to (1), we can just add an optional key parameter to record() and have EventCache dispatch both key and payload.

Could we set the topic to 'event' and let the listener decide which events to do something with?

Reply:
> Could we set the topic to 'event' and let the listener decide which events to do something with?

Done in revision 3